pallets/phala/src/lib.rs

```diff
@@ -297,28 +297,14 @@ decl_module! {
 			ensure!(Machine::contains_key(&machine_id), Error::<T>::BadMachineId);
 
 			let serialized_pk = Machine::get(machine_id).0;
-			let mut pk = [0u8; 33];
-			pk.copy_from_slice(&serialized_pk);
-			let pub_key = secp256k1::PublicKey::parse_compressed(&pk);
-			ensure!(pub_key.is_ok(), Error::<T>::InvalidPubKey);
+			if let Ok(()) = Self::verify_signature(serialized_pk, &transfer_data) {
+				T::TEECurrency::transfer(&Self::account_id(), &transfer_data.dest, transfer_data.amount, AllowDeath)
+					.map_err(|_| dispatch::DispatchError::Other("Can't transfer to chain"))?;
```

Comment:
```suggestion
			Self::verify_signature(serialized_pk, &transfer_data)?
			T::TEECurrency::transfer(&Self::account_id(), &transfer_data.dest, transfer_data.amount, AllowDeath)
				.map_err(|_| dispatch::DispatchError::Other("Can't transfer to chain"))?;
			...
```

Reply:
Please change like so, otherwise the error created by `ensure!` inside `verify_signature` won't be returned by `register_worker`